storage/files.js

```diff
@@ -72,326 +68,281 @@ function listFiles (name, callback) {
  *
  *   /a/1.txt
  *
- * @param {string} name The name of the bucket.
- * @param {string} prefix Filter results to objects whose names begin with this prefix.
- * @param {string} [delimiter] Results will contain only objects whose names, aside from the prefix, do not contain delimiter.
+ * @param {object} options Configuration options.
+ * @param {string} options.bucket The name of the bucket.
+ * @param {string} options.prefix Filter results to objects whose names begin
+ *     with this prefix.
+ * @param {string} [options.delimiter] Optional. Results will contain only
+ *     objects whose names, aside from the prefix, do not contain delimiter.
  * @param {function} cb The callback function.
  */
-function listFilesWithPrefix (name, prefix, delimiter, callback) {
-  if (!name) {
-    return callback(new Error('"name" is required!'));
-  } else if (!prefix) {
-    return callback(new Error('"prefix" is required!'));
-  }
+function listFilesByPrefix (options, callback) {
+  var bucket = storage.bucket(options.bucket);
 
-  var bucket = storage.bucket(name);
-  var options = {
-    prefix: prefix
+  var config = {
+    prefix: options.prefix
   };
-  if (delimiter && typeof delimiter === 'string') {
-    options.delimiter = delimiter;
+  if (options.delimiter && typeof options.delimiter === 'string') {
```

Comment:
Should we be checking types here? If `options.delimiter` isn't a string, I'm sure the user would want to be told that in a transparent way (e.g. an error message).